### ISSUE TITLE
Are there plans of including vs/workbench/contrib/remote/* source code

### ISSUE THREAD

**Issue by knopp** [microsoft/vscode-remote-release#391]
Remote workspaces currently don't work when building VSCode from source. I'm not asking about the actual extensions, I understand you want to keep them closed source, just the part that's missing from VSCode.

**Comment by mjbvz** [comment]
Duplicate of #30

**Comment by ericschultz** [comment]
That's not a duplicate. You're saying that if we build from source, we can't use your proprietary extension.

**Comment by knopp** [comment]
So it's not just name, telemetry and logos anymore, is it? This is a pretty significant feature gap between what you can get when you build the software yourself vs official release.

**Comment by egamma** [comment]
@knopp we are working on documenting the exact differences so that we are fully transparent.

//CC @chrisdias @chuxel @kieferrm

**Comment by knopp** [comment]
Thank you for the response. So after 3 and 1/2 years you're essentially making VSCode closed source. This is a rather big news and very sad one at that.

**Comment by ericschultz** [comment]
@knopp is [this](https://github.com/microsoft/vscode/tree/master/src/vs/workbench/contrib/remote) what is needed? It was just added 5 hours ago.

**Comment by knopp** [comment]
@ericschultz, indeed it is the file. But even with the file included the connection fails with "Unauthorized client refused." 

I'm guessing it has something to do with binary vsda module included in official VSCode build. So even with the file it's impossible to connect with custom build (it only works in development mode). I really don't understand why this isn't part of the extension, but vscode itself :-/

**Comment by egamma** [comment]
@knopp 

>Thank you for the response. So after 3 and 1/2 years you're essentially making VSCode closed source. This is a rather big news and very sad one at that.  

There was always a difference between `Code OSS` and `VS Code` the distro, pls see https://code.visualstudio.com/docs/supporting/faq#_what-is-the-difference-between-the-vscode-repository-and-the-microsoft-visual-studio-code-distribution

//fyi @chrisdias

**Comment by ericschultz** [comment]
@egamma maybe you should have told your copywriters about this if there was always a significant functional difference. Just picking a random date of January 19, 2018, the VSCode page said "open source" https://web.archive.org/web/20180119194122/https://code.visualstudio.com/

You folks effectively changed the rules out from under people. It sure seems like the warnings that senior open source folks have about Microsoft never being trustworthy in open source had some merit.

**Comment by egamma** [comment]
@ericschultz 

>You folks effectively changed the rules out from under people.

The licensing of VS Code (the distro) vs. Code OSS (the OSS project) has not changed and is the same since we open sourced the project. Please see this discussion from 2015 https://github.com/microsoft/vscode/issues/60#issuecomment-161792005.

What we did was to listen to feedback from the community and we have clarified the wording on the website.

**Comment by knopp** [comment]
@egamma, if I built vscode from source two months ago, I got pretty much the same functionality (sans some minor things like telemetry, branding and icons).  When I built it now, I get significantly crippled experience where major feature, remote connection doesn't work, even though I have the proprietary extensions installed. 

Yes, the code is there, and technically it is open source, but without the signer module it is essentially dead weight.
